From 1f1c794762720b9ae5e380248244946f71804d64 Mon Sep 17 00:00:00 2001
From: Greg Rose <gregory.v.rose@intel.com>
Date: Thu, 24 Jan 2013 04:54:48 +0000
Subject: [PATCH 724/858] ixgbevf: Make sure link status and speed are fetched

A recent change makes it necessary to set get_link_status to ensure that
the driver fetches the correct, refreshed value for link status and speed
when it has changed in the physical function device.

Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit aa19c2957b425f724efe93f2d5a5858b8aa613d7)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/intel/ixgbevf/ethtool.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/intel/ixgbevf/ethtool.c b/drivers/net/ethernet/intel/ixgbevf/ethtool.c
index 8f20704..c9d0c12 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ethtool.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ethtool.c
@@ -99,6 +99,7 @@ static int ixgbevf_get_settings(struct net_device *netdev,
  ecmd->transceiver = XCVR_DUMMY1;
  ecmd->port = -1;
 
+ hw->mac.get_link_status = 1;
  hw->mac.ops.check_link(hw, &link_speed, &link_up, false);
 
  if (link_up) {
-- 
1.8.3.1